src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -131,7 +131,7 @@ void Ekf::controlMagFusion(const imuSample &imu_sample)
 						     && mag_sample.mag.isAllFinite();
 
 		const bool starting_conditions_passing = continuing_conditions_passing
-				&& checkMagField(mag_sample.mag)
+				&& checkMagField(mag_sample.mag - _state.mag_B)
```

Comment:
Should we then rather have the `checkMagField` in `common_conditions_passing` ? Otherwise the bias can't be learned if "mag" fusion doesn't start.

Reply:
Yep, doesn't make sense otherwise. Thanks